### DISCUSSION TITLE
Feature Request: Allow uuid generation for random number generation (nonce, pkce, etc).

### DISCUSSION THREAD

**Discussion by william-will-angi** [panva/openid-client#635] [Ideas]
Our open id provider ( [keycloak](https://github.com/keycloak/keycloak) ) requires that nonces be passed in as uuids. Our current flow is throwing the error:

```
'nonce mismatch, expected `CVqEne3tzlU0J6fwt...`, got: afc784d4-d650-...'`
```

If there is a way we can force openid-client to generate uuid's for the nonce value, that would be ideal. Maybe some sort of optional override that could be passed in or something. I'm open to ideas and willing to contribute if appropriate here. 

Thanks!

**Comment by panva** [comment]
You can pass any value you want as the nonce parameter, you don't need to use one generated by the client.

**Reply by william-will-angi** [reply]
Sorry, I realized the place we would want this ability is in a downstream package (next-auth). Thank you for the quick response, and sorry again for the confusion!

Edit: In case anyone comes across this searching, I reposed this question to the next-auth community here: https://github.com/nextauthjs/next-auth/discussions/9197
